cesium/internal/index/domain.go

```diff
@@ -168,14 +174,18 @@ func (i *Domain) Stamp(
 			if !iter.Next() {
 				// exhausted
 				if continuous {
-					err = ErrDiscontinuous
+					err = errors.Wrapf(domain.ErrRangeNotFound, "cannot find stamp end with offset %d", offset)
 					return
 				}
 				approx = Between(iter.TimeRange().End, telem.TimeStampMax)
 				return
 			}
 			gap += iter.Len() / 8
 			if endOffset < gap {
+				err = r.Close()
```

Comment:
Why did you add all of these `Close` lines?

Reply:
need to close all domain/iterator level readers to avoid leaking them